components/fxa-client/src/migrator.rs

```diff
@@ -12,28 +12,37 @@ impl FirefoxAccount {
     /// * `session_token` - Hex-formatted session token.
     /// * `k_xcs` - Hex-formatted kXCS.
     /// * `k_sync` - Hex-formatted kSync.
+    /// * `in_place_migration` - If true then the provided 'session_token' will be migrated
+    ///                         instead of creating a new session token.
+    ///
     ///
     /// **💾 This method alters the persisted account state.**
     pub fn migrate_from_session_token(
         &mut self,
         session_token: &str,
         k_sync: &str,
         k_xcs: &str,
+        in_place_migration: bool,
     ) -> Result<()> {
         // if there is already a session token on account, we error out.
         if self.state.session_token.is_some() {
             return Err(ErrorKind::IllegalState("Session Token is already set.").into());
         }
-        // Trade our session token for a refresh token.
-        let duplicate_session = self
-            .client
-            .duplicate_session(&self.state.config, &session_token)?;
 
-        let duplicated_session_token = duplicate_session.session_token;
+        let mut migration_session_token = session_token.to_string();
+
+        if ! in_place_migration {
+            let duplicate_session = self
+                .client
+                .duplicate_session(&self.state.config, &session_token)?;
+
+            migration_session_token = duplicate_session.session_token;
+        }
+        // Trade our session token for a refresh token.
 
         let oauth_response = self.client.oauth_tokens_from_session_token(
             &self.state.config,
-            &duplicated_session_token,
+            &migration_session_token,
```

Comment:
What will happen to the device record attached to the existing `sessionToken` in this case?  I *think* it will be updated in place to also be associated with the new `refresh_token_id`, but it would be worth trying it out to make sure.  Ideally we would see the same entry in the "devices and apps" list before and after this migration.

Reply:
> Ideally we would see the same entry in the "devices and apps" list before and after this migration.

Yeap that's what I am seeing in my tests I think!